llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp

```diff
@@ -3421,6 +3421,23 @@ Value *InstCombinerImpl::foldAndOrOfICmps(ICmpInst *LHS, ICmpInst *RHS,
   return foldAndOrOfICmpsUsingRanges(LHS, RHS, IsAnd);
 }
 
+Value *foldAorBZero(BinaryOperator &I, InstCombiner::BuilderTy &Builder) {
+  Value *Op0 = I.getOperand(0);
```

Comment:
Since this is only for `or` at the moment, think an assert of `I.getOpcode` would be useful to avoid future misreads.